source/geometries/Cigar.cc

```diff
@@ -0,0 +1,512 @@
+    G4MaterialPropertiesTable *this_fiber_optical = nullptr;
+    if (fiber_type_ == "Y11") {
+      this_fiber_optical = opticalprops::Y11();
+    } else if (fiber_type_ == "B2") {
+      this_fiber_optical = opticalprops::B2();
+    } else {
+      G4Exception("[Cigar]", "Construct()",
+                  FatalException, "Invalid fiber type, must be Y11 or B2");
+    }
+
+
+    G4Material *this_coating = nullptr;
+    G4MaterialPropertiesTable *this_coating_optical = nullptr;
+    if (coated_) {
+      if (coating_ == "TPB") {
+        this_coating = materials::TPB();
+        this_coating_optical = opticalprops::TPB();
+      } else if (coating_ == "TPH") {
+        this_coating = materials::TPH();
+        this_coating_optical = opticalprops::TPH();
+      } else {
+        G4Exception("[Cigar]", "Construct()",
+                    FatalException, "Invalid coating, must be TPB or TPH");
+      }
+    }
+
+    fiber_ = new GenericWLSFiber(fiber_type_, true, true, fiber_diameter_, cigar_length_ + 7 * cm, true, coated_, this_coating, this_fiber, true);
+
+    // WORLD /////////////////////////////////////////////////
+
+    G4Material* world_mat = nullptr;
+
+    if (gas_ == "Ar") {
+      world_mat = materials::GAr(pressure_);
+      world_mat->SetMaterialPropertiesTable(opticalprops::GAr(1. / (68 * eV)));
+
+    } else if (gas_ == "Xe") {
+      world_mat = materials::GXe(pressure_);
+      // world_mat->SetMaterialPropertiesTable(opticalprops::GXe(pressure_, 273.15, 250, (1.0E9)));
+      world_mat->SetMaterialPropertiesTable(opticalprops::GXe());
+    } else {
+      G4Exception("[Cigar]", "Construct()",
+            FatalException, "Invalid gas, must be Ar or Xe");
+    }
+
+    G4Box* world_solid_vol =
+     new G4Box("WORLD", world_xy_/2., world_xy_/2., world_z_/2.);
+
+    G4LogicalVolume* world_logic_vol =
+      new G4LogicalVolume(world_solid_vol, world_mat, "WORLD");
+    world_logic_vol->SetVisAttributes(G4VisAttributes::GetInvisible());
+    GeometryBase::SetLogicalVolume(world_logic_vol);
+
+    // TEFLON PANELS ////////////////////////////////////////////
+    G4double panel_width = 2.5 * mm;
+    G4double extra_width = (5) * mm;
+    // G4Box* teflon_panel_top =
+    //   new G4Box("TEFLON_PANEL_TOP", cigar_width_ / 2 + extra_width + panel_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_top =
+      new G4Box("TEFLON_PANEL_TOP", cigar_width_ / 2 + extra_width + panel_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_side =
+      new G4Box("TEFLON_PANEL_SIDE", cigar_width_ / 2 + extra_width, panel_width / 2, cigar_length_ / 2);
+    G4Box* teflon_panel_close =
+      new G4Box("TEFLON_PANEL_CLOSE", cigar_width_ / 2 + extra_width+panel_width, cigar_width_ / 2 + extra_width+panel_width, panel_width / 2);
+    G4Material* teflon = G4NistManager::Instance()->FindOrBuildMaterial("G4_TEFLON");
+    teflon->SetMaterialPropertiesTable(opticalprops::PTFE());
```

Comment:
The optical properties are assigned to the surface, later on. Remove them from the teflon bulk volume.